llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -18142,6 +18165,26 @@ static SDValue foldReduceOperandViaVQDOT(SDValue InVec, const SDLoc &DL,
       !InVec.getValueType().getVectorElementCount().isKnownMultipleOf(4))
     return SDValue();
 
+  // Recurse through adds (since generic dag canonicalizes to that
+  // form).
+  if (InVec->getOpcode() == ISD::ADD) {
```

Comment:
Just wondering, will we ever have to deal with disjoint ors here? Nothing that needs to be done for this patch.